tests/posts/test_posts.py

```diff
@@ -192,10 +192,44 @@ def test_get_commments_invalid_step(self, shared_variables, session):
         pass
 
     def test_delete_post_valid(self, shared_variables, session):   
-        pass
+        url = f"{self.BASE_URL}/posts/{shared_variables['current_post_id']}/delete"
+        response = session.delete(url, cookies=session.cookies.get_dict())
 
-    def test_delete_post_invalid_post_id(self, shared_variables, session):   
-        pass
+        assert response.status_code == 200, self.buildErrorMessage(
+            response.status_code, response.content
+        )
 
-    def test_delete_post_not_post_owner(self, shared_variables, session):   
-        pass
+        response = session.get(url, cookies=session.cookies.get_dict())
+        assert response.status_code == 404, self.buildErrorMessage(
+            response.status_code, response.content
+        )
+
+    def test_delete_post_invalid_post_id(self, session):
+        url = f"{self.BASE_URL}/posts/invalidPostId/delete"
+        response = session.delete(url, cookies=session.cookies.get_dict())
+
+        assert response.status_code == 404, self.buildErrorMessage(
+            response.status_code, response.content
+        )
+        
+    def test_delete_post_not_post_owner(self, shared_variables, session):
+        new_post_data = {
+            "title": "Test Post",
+            "content": "This is a test post."
+        }
+        new_post_response = session.post(f"{self.BASE_URL}/posts", json=new_post_data)
+
+        if "error" in new_post_response.json():
+            assert new_post_response.json()["error"] == "No session token", f"Unexpected error: {new_post_response.json()}"
```

Comment:
This needs to be refactored. 
* It's not using self.buildErrorMessage() like the other tests.
* It doesn't check the correct status code (should be **explicitly** checking for 403).